tests/monitoring_integration_test.py

```diff
@@ -137,352 +163,226 @@ def testUptimeCheckAlertingPolicyNotificationChannel(self):
+        except (IndexError, ZeroDivisionError) as e:
+            # data not found. Return None
+            pass
+        return ratio
+
+    def get_service_latency(self, service_name, period_seconds=1200):
+        """
+        Calculates the latency data for a service
+
+        Args:
+            service_name (str): the name of the service to query
+            period_seconds (int): the number of seconds back to read metrics for
+        Returns:
+            the calculated average latency of the service in miliseconds as a float
+        """
+        filter_ = f'metric.type = "istio.io/service/server/response_latencies" AND resource.labels.canonical_service_name = "{service_name}"'
+        aggregation = {
+            "alignment_period": {"seconds": period_seconds},
+            "per_series_aligner": monitoring_v3.types.Aggregation.Aligner.ALIGN_DELTA,
+            "cross_series_reducer": monitoring_v3.types.Aggregation.Reducer.REDUCE_MEAN,
+        }
+        results = self._get_metric_data(filter_, aggregation, period_seconds)
+        results = list(results)
+        latency = None
+        try:
+            latency = results[0].points[0].value.double_value
+        except IndexError:
+            # data not found. Return None
+            pass
+        return latency
+
+    def test_services_created(self):
+        """
+        Ensure that hipstersop Istio services have been picked up by Cloud Monitoring
+        """
+        for service_name in _services_short:
+            istio_service_name = self.getIstioService(service_name)
+            full_name = self.client.service_path(self.project_id, istio_service_name)
+            result = self.client.get_service(full_name)
+            self.assertEqual(result.display_name, service_name, f"{service_name} not found")
+            print(f"✅  {service_name} canonical service created")
+
+    def test_slos_created(self):
+        """
+        Ensure that SLOs have been added to Istio services
+        """
+        for service_name in _services_short:
+            istio_service_name = self.getIstioService(service_name)
+            for slo_type in ['latency', 'availability']:
+                slo_id = f"{service_name}-{slo_type}-slo"
+                slo_name_full = self.client.service_level_objective_path(
+                    self.project_id, istio_service_name, slo_id)
+                result = self.client.get_service_level_objective(slo_name_full)
+                self.assertIsNotNone(result, f"{service_name} {slo_type} SLO not found")
+                print(f"✅  {service_name} {slo_type} SLO created")
+
+    def test_availability_slos_passing(self):
+        """
+        Ensure that availability is at expected levels to pass the SLO
+        """
+        for service_name in _services_short:
+            # get SLO
+            istio_service_name = self.getIstioService(service_name)
+            slo_id = f"{service_name}-availability-slo"
```

Comment:
I see that this piece of code also exists in `test_latency_slos_passing()` - can you please take a look and see if you can move reusable code into separate functions?

Reply:
Done